stan/math/prim/arr/functor/map_rect_mpi.hpp

```diff
@@ -0,0 +1,147 @@
+#pragma once
+
+#include <vector>
+#include <set>
+
+#include <stan/math/prim/mat/fun/to_array_1d.hpp>
+#include <stan/math/prim/arr/functor/mpi_command.hpp>
+#include <stan/math/prim/arr/functor/mpi_cluster.hpp>
+
+namespace stan {
+  namespace math {
+
+    namespace internal {
+      // forward declare
+      void
+      distribute_map_rect_data(const std::vector<std::vector<double> >& x_r,
+                               const std::vector<std::vector<int> >& x_i,
+                               const std::size_t uid);
+
+
+      // data distribution needs to go to prim
+      struct distributed_map_rect_data {
+        distributed_map_rect_data(std::size_t uid) : uid_(uid), N_(-1) {}
+        std::size_t uid_;
+        std::size_t N_;
+        std::vector<std::vector<double> > x_r_;
+        std::vector<std::vector<int> > x_i_;
+        static void distributed_apply() {
+          // called on workers to register data
+          distribute_map_rect_data(std::vector<std::vector<double> >(), std::vector<std::vector<int> >(), 0);
+        }
+      };
+
+      typedef boost::serialization::singleton<std::map<std::size_t,const distributed_map_rect_data> > distributed_data;
+
+      void
+      distribute_map_rect_data(const std::vector<std::vector<double> >& x_r,
+                               const std::vector<std::vector<int> >& x_i,
+                               const std::size_t uid) {
+        
+        boost::mpi::communicator world;
+        
+        const std::size_t W = world.size();
+        const std::size_t R = world.rank();
+
+        // first check if uid is already registered
+        if(distributed_data::get_const_instance().find(uid) == distributed_data::get_const_instance().cend()) {
+          //std::cout << "Job " << R << " registers data..." << std::endl;
```

Comment:
Is this stubbed out for now or old debug code?

Reply:
yep, this is old debug code. Not meant to stay.